### PR TITLE
WrapGL machine: return both high and low values

### PR DESCRIPTION
**Pull request by georgwiese**
In Leo's use-case of converting a Poseidon hash to a rust `u64`, we actually want both the high and low 32 bit of the field element being wrapped. I added this to the WrapGL machine.

Fruit for thought: suppose the caller only wants the higher or lower half, would it be possible to optimize-away all the columns I added now?

### CODE REVIEW

**Comment by georgwiese** [comment]
Not sure is wrap is still a great name?

**Comment by leonardoalt** [comment]
yea let's call it Split64?

**Comment by leonardoalt** [comment]
makes sense!

File: std/split/split_gl.asm [hunk 1]
```diff
@@ -22,9 +22,11 @@ machine WrapGL(RESET, operation_id) {
 
     // 2. Build the output, packing the least significant 4 byte into
     //    a field element
-    col witness output;
-    col fixed FACTOR_OUTPUT = [0x100, 0x10000, 0x1000000, 0, 0, 0, 0, 1]*;
-    output' = (1 - RESET) * output + bytes * FACTOR_OUTPUT;
+    col witness output_low, output_high;
+    col fixed FACTOR_OUTPUT_LOW = [0x100, 0x10000, 0x1000000, 0, 0, 0, 0, 1]*;
+    col fixed FACTOR_OUTPUT_HIGH = [0, 0, 0, 1, 0x100, 0x10000, 0x1000000, 0]*;
+    output_low' = (1 - RESET) * output_low + bytes * FACTOR_OUTPUT_LOW;
```
Comment (leonardoalt) [inline std/split/split_gl.asm:28]: so bytes here is the full 8 bytes, right?

Comment (georgwiese) [reply std/split/split_gl.asm:28]: Every entry in the bytes column is one byte, but there are 8 rows in a block, so it lists the full 8-byte decomposition of the input.

Comment (leonardoalt) [reply std/split/split_gl.asm:28]: ok

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**